src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/DefaultJsonTypeInfoResolver.cs

```diff
@@ -29,9 +29,10 @@ public DefaultJsonTypeInfoResolver() : this(mutable: true)
         [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
         private DefaultJsonTypeInfoResolver(bool mutable)
         {
-            JsonSerializerOptions.RootConverters();
             _mutable = mutable;
-            Modifiers = new ModifierCollection(this);
+
+            s_defaultFactoryConverters ??= GetDefaultFactoryConverters();
+            s_defaultSimpleConverters ??= GetDefaultSimpleConverters();
```

Comment:
I'm wondering if it's better to have that in static constructor or just field initializer. If you don't touch DefaultJsonTypeInfoResolver it won't instantiate it anyway

Reply:
Initializing converters in the static constructor would end up rooting the converters. The type exposes a bunch of static internal methods which are used in non-reflection scenaria.